From 28875e1a96d852838a1ef99bc75a22cf2f831cf4 Mon Sep 17 00:00:00 2001
From: Ken Tseng <ken.tseng@realtek.com>
Date: Tue, 8 Mar 2022 14:27:46 +0800
Subject: [PATCH 34/54] 815: perf: Add realtek perf driver

---
 drivers/perf/Kconfig  | 1 +
 drivers/perf/Makefile | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/perf/Kconfig b/drivers/perf/Kconfig
index 130327ff0..9cb17fff3 100644
--- a/drivers/perf/Kconfig
+++ b/drivers/perf/Kconfig
@@ -131,5 +131,6 @@ config ARM_SPE_PMU
 	  the CPU pipeline and reports this via the perf AUX interface.
 
 source "drivers/perf/hisilicon/Kconfig"
+source "drivers/perf/realtek/Kconfig"
 
 endmenu
diff --git a/drivers/perf/Makefile b/drivers/perf/Makefile
index 5365fd56f..6a72b7295 100644
--- a/drivers/perf/Makefile
+++ b/drivers/perf/Makefile
@@ -13,3 +13,4 @@ obj-$(CONFIG_QCOM_L3_PMU) += qcom_l3_pmu.o
 obj-$(CONFIG_THUNDERX2_PMU) += thunderx2_pmu.o
 obj-$(CONFIG_XGENE_PMU) += xgene_pmu.o
 obj-$(CONFIG_ARM_SPE_PMU) += arm_spe_pmu.o
+obj-$(CONFIG_ARCH_REALTEK) += realtek/
-- 
2.17.1

