From 5508155446a9030bb805549c8f115a29bc0444ae Mon Sep 17 00:00:00 2001
From: Ken Tseng <ken.tseng@realtek.com>
Date: Tue, 8 Mar 2022 14:32:35 +0800
Subject: [PATCH 39/54] 820: regulator: Add REGULATOR_APW888X

---
 drivers/regulator/Kconfig  | 18 ++++++++++++++++++
 drivers/regulator/Makefile |  3 +++
 2 files changed, 21 insertions(+)

diff --git a/drivers/regulator/Kconfig b/drivers/regulator/Kconfig
index 020a00d66..261c3aa16 100644
--- a/drivers/regulator/Kconfig
+++ b/drivers/regulator/Kconfig
@@ -1279,5 +1279,23 @@ config REGULATOR_QCOM_LABIBB
 	  boost regulator and IBB can be used as a negative boost regulator
 	  for LCD display panel.
 
+config REGULATOR_APW888X_CORE
+	tristate
+
+config REGULATOR_APW8889
+	tristate "ANPEC APW8889 Power regulators"
+	depends on MFD_APW8889_I2C
+	select REGULATOR_APW888X_CORE
+	help
+	  This driver provides support for the voltage regulators on the
+	  Anpec APW8889.
+
+config REGULATOR_APW8886
+	tristate "ANPEC APW8886 Power regulators"
+	depends on MFD_APW8886_I2C
+	select REGULATOR_APW888X_CORE
+	help
+	  This driver provides support for the voltage regulators on the
+	  Anpec APW8886.
 endif
 
diff --git a/drivers/regulator/Makefile b/drivers/regulator/Makefile
index 6ebae5162..59d6bd6e8 100644
--- a/drivers/regulator/Makefile
+++ b/drivers/regulator/Makefile
@@ -157,5 +157,8 @@ obj-$(CONFIG_REGULATOR_WM831X) += wm831x-ldo.o
 obj-$(CONFIG_REGULATOR_WM8350) += wm8350-regulator.o
 obj-$(CONFIG_REGULATOR_WM8400) += wm8400-regulator.o
 obj-$(CONFIG_REGULATOR_WM8994) += wm8994-regulator.o
+obj-$(CONFIG_REGULATOR_APW888X_CORE) += apw888x-regulator-core.o
+obj-$(CONFIG_REGULATOR_APW8889) += apw8889-regulator.o
+obj-$(CONFIG_REGULATOR_APW8886) += apw8886-regulator.o
 
 ccflags-$(CONFIG_REGULATOR_DEBUG) += -DDEBUG
-- 
2.17.1

