--- a/src/utils/mount.ecryptfs_private.c
+++ b/src/utils/mount.ecryptfs_private.c
@@ -224,6 +224,7 @@ err:
 
 static int check_cwd_f_type()
 {
+	struct statfs buf;
 	/**
 	 * This is *not* a list of compatible lower filesystems list for
 	 * eCryptfs. This is a list of filesystems that we reasonably expect to
@@ -235,7 +236,7 @@ static int check_cwd_f_type()
 	 * deceive other programs with a crafted /proc/self/*. See
 	 * https://launchpad.net/bugs/1530566 for more details.
 	 */
-	__SWORD_TYPE f_type_whitelist[] = {
+    typeof(buf.f_type) f_type_whitelist[] = {
 		0x61756673 /* AUFS_SUPER_MAGIC */,
 		0x9123683E /* BTRFS_SUPER_MAGIC */,
 		0x00C36400 /* CEPH_SUPER_MAGIC */,
@@ -259,7 +260,6 @@ static int check_cwd_f_type()
 		0x58465342 /* XFS_SB_MAGIC */,
 		0x2FC12FC1 /* ZFS_SUPER_MAGIC */,
 	};
-	struct statfs buf;
 	size_t i, whitelist_len;
 
 	if (statfs(".", &buf) != 0) {
